patient-db/src/main/java/com/hartwig/hmftools/patientdb/amber/LoadAmberData.java

```diff
@@ -97,32 +97,41 @@ public static void main(@NotNull String[] args) throws ParseException, IOExcepti
 
     public static void processSample(final AmberSample sample, final DatabaseAccess dbAccess)
     {
-        LOGGER.info("comparing with existing samples");
-
-        final List<AmberSample> allSamples = dbAccess.readAmberSamples();
-        final List<AmberMapping> sampleMappings = new ArrayList<>();
-        for(AmberSample other : allSamples)
+        try
         {
-            if(!other.sampleId().equals(sample.sampleId()))
+            dbAccess.context().execute("LOCK TABLES amberPatient WRITE, amberSample WRITE, amberMapping WRITE");
```

Comment:
Maybe add some logging before locking and unlocking tables? That way it will be more clear why a concurrently run Isofox loader "hangs" for a while.